pkg/resources/task_acceptance_test.go

```diff
@@ -1,564 +1,970 @@
+						HasDatabaseString(id.DatabaseName()).
+						HasSchemaString(id.SchemaName()).
+						HasNameString(id.Name()).
+						HasEnabledString(r.BooleanFalse).
+						HasWarehouseString("").
+						HasScheduleString("").
+						HasConfigString("").
+						HasAllowOverlappingExecutionString(r.BooleanDefault).
+						HasErrorIntegrationString("").
+						HasCommentString("").
+						HasFinalizeString("").
+						HasAfterIds().
+						HasWhenString("").
+						HasSqlStatementString(statement),
+					resourceshowoutputassert.TaskShowOutput(t, basicConfigModel.ResourceReference()).
+						HasCreatedOnNotEmpty().
+						HasName(id.Name()).
+						HasIdNotEmpty().
+						HasDatabaseName(id.DatabaseName()).
+						HasSchemaName(id.SchemaName()).
+						HasOwner(currentRole.Name()).
+						HasComment("").
+						HasWarehouse("").
+						HasSchedule("").
+						HasPredecessors().
+						HasState(sdk.TaskStateSuspended).
+						HasDefinition(statement).
+						HasCondition("").
+						HasAllowOverlappingExecution(false).
+						HasErrorIntegration("").
+						HasLastCommittedOn("").
+						HasLastSuspendedOn("").
+						HasOwnerRoleType("ROLE").
+						HasConfig("").
+						HasBudget("").
+						HasTaskRelations(sdk.TaskRelations{}),
+				),
+			},
+			// Set
+			{
+				Config: config.FromModel(t, completeConfigModel),
+				Check: assert.AssertThat(t,
+					resourceassert.TaskResource(t, completeConfigModel.ResourceReference()).
+						HasFullyQualifiedNameString(id.FullyQualifiedName()).
+						HasDatabaseString(id.DatabaseName()).
+						HasSchemaString(id.SchemaName()).
+						HasNameString(id.Name()).
+						HasEnabledString(r.BooleanTrue).
+						HasWarehouseString(warehouse.ID().Name()).
+						HasScheduleString("5 MINUTES").
+						HasConfigString(expectedTaskConfig).
+						HasAllowOverlappingExecutionString(r.BooleanTrue).
+						HasErrorIntegrationString(errorNotificationIntegration.ID().Name()).
+						HasCommentString(comment).
+						HasFinalizeString("").
+						HasAfterIds().
+						HasWhenString(condition).
+						HasSqlStatementString(statement),
+					resourceshowoutputassert.TaskShowOutput(t, completeConfigModel.ResourceReference()).
+						HasCreatedOnNotEmpty().
+						HasName(id.Name()).
+						HasIdNotEmpty().
+						HasDatabaseName(id.DatabaseName()).
+						HasSchemaName(id.SchemaName()).
+						HasOwner(currentRole.Name()).
+						HasWarehouse(warehouse.ID().Name()).
+						HasComment(comment).
+						HasSchedule("5 MINUTES").
+						HasPredecessors().
+						HasState(sdk.TaskStateStarted).
+						HasDefinition(statement).
+						HasCondition(condition).
+						HasAllowOverlappingExecution(true).
+						HasErrorIntegration(errorNotificationIntegration.ID().Name()).
+						HasLastCommittedOnNotEmpty().
+						HasLastSuspendedOn("").
+						HasOwnerRoleType("ROLE").
+						HasConfig(expectedTaskConfig).
+						HasBudget("").
+						HasTaskRelations(sdk.TaskRelations{}),
+				),
```

Comment:
Pls add a test case with external changes.

Reply:
+1

Reply:
Added test cases for Finalize and After fields, added TODO for next pr to see what also could be tested